docs/deploy/Deploy-AppInsights.bicep

```diff
@@ -0,0 +1,85 @@
+@description('Location where the resources are deployed. For list of Azure regions where application insights is available, see [Products available by region](https://azure.microsoft.com/en-us/explore/global-infrastructure/products-by-region/?products=monitor).')
+@allowed([
+  'australiacentral'
+  'australiaeast'
+  'australiasoutheast'
+  'brazilsouth'
+  'canadacentral'
+  'canadaeast'
+  'centralindia'
+  'centralus'
+  'chinaeast2'
+  'chinanorth3'
+  'eastasia'
+  'eastus'
+  'eastus2'
+  'francecentral'
+  'germanywestcentral'
+  'israelcentral'
+  'italynorth'
+  'japaneast'
+  'japanwest'
+  'koreacentral'
+  'northcentralus'
+  'northeurope'
+  'norwayeast'
+  'polandcentral'
+  'qatarcentral'
+  'southafricanorth'
+  'southcentralus'
+  'southeastasia'
+  'southindia'
+  'swedencentral'
+  'switzerlandnorth'
+  'uaenorth'
+  'uksouth'
+  'ukwest'
+  'westeurope'
+  'westus'
+  'westus2'
+  'westus3'
+])
+param location string
+
+@description('Name of the container apps environment.')
+param envName string
+
+@description('The resource ID of the log analytics workspace to link to the application insights instance.')
+param logAnalyticsWorkspaceId string
+
+// Deploy application insights for receiving azure monitor telemetry
+var applicationInsightsName = '${envName}-ai'
+resource applicationInsights 'Microsoft.Insights/components@2020-02-02' = {
+  name: applicationInsightsName
+  location: location
+  kind: 'web'
+  properties: {
+    Application_Type: 'web'
+    WorkspaceResourceId: logAnalyticsWorkspaceId
+    DisableLocalAuth: true
+  }
+}
+
+// Create user-assigned managed identity to authenticate with Application Insights
+var applicationInsightsUAMIName = '${applicationInsightsName}-mi'

```

Comment:
Just curious, does assigning the app's system assigned MI to App insights not work?